### ISSUE TITLE
Bug: There are multiple custom text nodes within a single paragraph node, where when you enter it in Korean, the cursor moves to the front of the node and the text is entered.

### ISSUE THREAD

**Issue by yunseop-dev** [facebook/lexical#3899]
Lexical version: 7.0

## Steps To Reproduce
After updating from version 0.6.5 to version 7.0+, the following occurs:

The following bug occurs in the latest version of the Chrome browser.

I am attaching a screenshot together.

## The current behavior
There are multiple custom text nodes within a single paragraph node, where when you enter it in Korean, the cursor moves to the front of the node and the text is entered.

There is no bug when typing text in English.

## The expected behavior
Expect normal text input

## screenshots
![Feb-14-2023 17-07-15](https://user-images.githubusercontent.com/63530844/218676074-2391baf8-a4bc-4fc6-82ed-d53dfda60923.gif)

**Comment by trueadm** [comment]
Can you maybe help us narrow down the issue by finding what version was last good before this regression?

**Comment by yunseop-dev** [comment]
> Can you maybe help us narrow down the issue by finding what version was last good before this regression?

@trueadm  Yes, so I tried tracking down from 0.8.0 version, but it works fine in 0.6.5 version, and there is a problem in more versions.

**Comment by yunseop-dev** [comment]
@trueadm I found it late, and the following error also occurs.

`Uncaught DOMException: Failed to execute 'setBaseAndExtent' on 'Selection': The offset 5 is larger than the node's length (4).`

**Comment by yunseop-dev** [comment]
Hi, is there any update? It seemed that the contents related to Korean were merged with the 0.9.1 version, right?

**Comment by mayrang** [comment]
Tested on current main (playground, Chrome, macOS, Korean IME) with multiple TextNodes in a single paragraph — bold/normal/italic boundaries. Korean input works correctly at all boundary positions, no cursor jump.

Also tested on v0.7.0 (the reported version) with the same setup — couldn't reproduce there either. The original report mentions custom text nodes with overridden createDOM, which may have been the specific trigger, but without a reproduction sample it's hard to narrow down further.

There have been several composition-related fixes since v0.7.0 (notably the compositionKey scoping in #3679 and the targetRange guards). If the issue is still occurring on a current version with specific custom TextNode subclasses, a minimal reproduction would help reopen.
